File: 264. Ugly Number II/Python/Solution.py
class Solution(object):
    def nthUglyNumber(self, n):
        """
        :type n: int
        :rtype: int
        """
        index_2 = -1
        index_3 = -1
        index_5 = -1
        ugly = []
        factor_2 = 1
        factor_3 = 1
        factor_5 = 1
        for i in range(n):
            ugly.append(min(factor_2, factor_3, factor_5))
            if ugly[i] == factor_2:
                index_2 += 1
                factor_2 = ugly[index_2] * 2
            if ugly[i] == factor_3:
                index_3 += 1
                factor_3 = ugly[index_3] * 3
            if ugly[i] == factor_5:
                index_5 += 1
                factor_5 = ugly[index_5] * 5
        return ugly[-1]

File: 264. Ugly Number II/Java/Solution.java
public class Solution {
    public int nthUglyNumber(int n) {
        int index2 = -1, index3 = -1, index5 = -1;
        int[] ugly = new int[n];
        int factor2 = 1, factor3 = 1, factor5 = 1;
        for (int i = 0; i < n; i++) {
            ugly[i] = Math.min(Math.min(factor2, factor3), factor5);
            if (ugly[i] == factor2) {
                index2++;
                factor2 = ugly[index2] * 2;
            }
            if (ugly[i] == factor3) {
                index3++;
                factor3 = ugly[index3] * 3;
            }
            if (ugly[i] == factor5) {
                index5++;
                factor5 = ugly[index5] * 5;
            }
        }
        return ugly[n - 1];
    }
}